### PR TITLE
[compute] handle case of no hive or impala compute

### PR DESCRIPTION
**Pull request by amitsrivastava**
Currently we show the Hive or Impala editors even when there isn't a single compute available for that interpreter. This leads to unnecessary errors.

This commit adds an additional check for presence of computes while returning the list of editors in get_config call.

Change-Id: Id3837cc3b81da5f54d7539b89a6a9ae2f4a122b1

### CODE REVIEW

File: desktop/libs/notebook/src/notebook/conf.py [hunk 1]
```diff
@@ -152,6 +157,16 @@ def get_ordered_interpreters(user=None):
     for i in interpreters
   ]
 
+def computes_for_dialect(dialect, user):
+  from beeswax.models import Namespace
```
Comment (agl29) [inline desktop/libs/notebook/src/notebook/conf.py:161]: why here? is it due to Cyclic dependency? if Yes can you please add a comment # Cyclic dependency after import.

Comment (amitsrivastava) [reply desktop/libs/notebook/src/notebook/conf.py:161]: yes, this is due to cyclic deps. Will add a comment.

**Review by agl29 — APPROVED** [review]
LGTM!

**Review by ranade1 — APPROVED** [review]
+1

**Pull request merged**